sycamore/transforms/partition.py

```diff
@@ -49,6 +49,91 @@ def partition(self, document: Document) -> Document:
         pass
 
 
+class UnstructuredPPTXPartitioner(Partitioner):
+    """
+    UnstructuredPPTXPartitioner utilizes open-source Unstructured library to extract structured elements from
+    unstructured PPTX files.
+
+    Args:
+        include_page_breaks: Whether to include page breaks as separate elements.
+        strategy: The partitioning strategy to use ("auto" for automatic detection).
+        infer_table_structure: Whether to infer table structures in the document.
+        ocr_languages: The languages to use for OCR. Default is "eng" (English).
+        max_partition_length: The maximum length of each partition (in characters).
+        include_metadata: Whether to include metadata in the partitioned elements.
+
+    Example:
+         .. code-block:: python
+
+            pptx_partitioner = UnstructuredPPTXPartitioner(
+                include_page_breaks=True,
+                strategy="auto",
+                infer_table_structure=True,
+                ocr_languages="eng",
+                max_partition_length=2000,
+                include_metadata=True,
+            )
+
+            context = sycamore.init()
+            pdf_docset = context.read.binary(paths, binary_format="pptx")
+                .partition(partitioner=pptx_partitioner)
+
+    """
+
+    @staticmethod
+    def to_element(dict: dict[str, Any]) -> Element:
+        text = dict.pop("text")
+        if isinstance(text, str):
+            binary = text.encode("utf-8")
+        else:
+            binary = text
+            text = str(binary, "utf-8")
+
+        element = Element()
+        element.type = dict.pop("type", "unknown")
+        element.binary_representation = binary
+        element.text_representation = text
+        element.properties.update(dict.pop("metadata"))
+        element.properties.update(dict)
+
+        return element
+
+    def __init__(
+        self,
+        include_page_breaks: bool = False,
+        include_metadata: bool = True,
+        include_slide_notes: bool = False,
+        chunking_strategy: Optional[str] = None,
+        **kwargs
+    ):
+        self._include_page_breaks = include_page_breaks
+        self._include_metadata = include_metadata
+        self._include_slide_notes = include_slide_notes
+        self._chunking_strategy = chunking_strategy
+        self._kwargs = kwargs
+
+    def partition(self, document: Document) -> Document:
+        from unstructured.partition.pptx import partition_pptx
```

Comment:
Question rather than a comment -- is there a Ray reason for importing these as part of the method rather than at the top?

Reply:
Nop! Just to bring in the dependency as needed.